app/models/attempt.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class Attempt < ApplicationRecord
+  include Comparable
+
+  default_scope { order(:attempt_number) }
+
+  belongs_to :result
+
+  validates :value, presence: true
+  validates :value, numericality: { only_integer: true }
+  validates :attempt_number, numericality: { only_integer: true }
+  validates :result_id, uniqueness: { scope: [:attempt_number] }
+
+  def <=>(other)
+    value <=> other.value
+  end
```

Comment:
Just curious, does `delegate :<=>, to: :value` work?

Reply:
sadly not. 
`comparison of Attempt with Attempt failed (ArgumentError)
`